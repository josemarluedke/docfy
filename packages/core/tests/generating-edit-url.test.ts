import Docfy from '../src';
import path from 'path';

const root = path.resolve(__dirname, './__fixtures__/monorepo');

test('it should correctly generate edit urls', async () => {
  const docfy = new Docfy({
    repository: {
      url: 'https://github.com/user/repo',
      editBranch: 'dev',
    },
  });
  const result = await docfy.run([
    {
      root,
      urlPrefix: 'docs',
      urlSchema: 'manual',
      pattern: '**/*.md',
      ignore: ['**/package2/**/*.md'],
    },
    {
      root,
      urlPrefix: 'docs',
      urlSchema: 'manual',
      pattern: '**/package2/**/*.md',
      repository: {
        url: 'https://bitbucket.org/user/repo',
      },
    },
  ]);

  const urls = [];
  result.content.forEach(page => {
    urls.push([page.source, page.meta.editUrl]);
  });

  expect(urls).toMatchSnapshot();
});
