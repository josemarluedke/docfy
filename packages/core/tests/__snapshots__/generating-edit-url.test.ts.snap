// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should correctly generate edit urls 1`] = `
[
  [
    "README.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/README.md",
  ],
  [
    "docs/about.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/docs/about.md",
  ],
  [
    "docs/custom-url.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/docs/custom-url.md",
  ],
  [
    "docs/how-it-works.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/docs/how-it-works.md",
  ],
  [
    "docs/html.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/docs/html.md",
  ],
  [
    "docs/math.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/docs/math.md",
  ],
  [
    "docs/static-asset.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/docs/static-asset.md",
  ],
  [
    "docs/table.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/docs/table.md",
  ],
  [
    "packages/package1/components/button.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/packages/package1/components/button.md",
  ],
  [
    "packages/package1/docs/intro.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/packages/package1/docs/intro.md",
  ],
  [
    "packages/package1/components/form/index.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/packages/package1/components/form/index.md",
  ],
  [
    "packages/package1/components/form/something-else.md",
    "https://github.com/user/repo/edit/dev/packages/core/tests/__fixtures__/monorepo/packages/package1/components/form/something-else.md",
  ],
  [
    "packages/package2/docs/overview.md",
    "https://bitbucket.org/user/repo/src/dev/packages/core/tests/__fixtures__/monorepo/packages/package2/docs/overview.md?mode=edit&spa=0&at=dev&fileviewer=file-view-default",
  ],
  [
    "packages/package2/docs/styles.md",
    "https://bitbucket.org/user/repo/src/dev/packages/core/tests/__fixtures__/monorepo/packages/package2/docs/styles.md?mode=edit&spa=0&at=dev&fileviewer=file-view-default",
  ],
]
`;
